lib/views/location.dart

```diff
@@ -0,0 +1,108 @@
+import 'package:flutter/material.dart';
+import 'package:one_app/utils/location_time.dart';
+import 'package:flutter_spinkit/flutter_spinkit.dart';
+
+class Location extends StatefulWidget {
+  final LocationTime location;
+  const Location({super.key, required this.location});
+
+  @override
+  State<Location> createState() => _LocationState();
+}
+
+class _LocationState extends State<Location> {
+  void setUpTime() async {
+    await widget.location.getTime();
+    setState(() {});
+  }
+
+  @override
+  void initState() {
+    super.initState();
+
+    setUpTime();
```

Comment:
you also need to implement `didUpdateWidget` to check in there whether `location` has changed. If it has, you will likely have to run setUpTime again...